### PR TITLE
Cleanup extensions dependencies

### PR DESCRIPTION
**Pull request by rijkvanzanten**
## Scope

What's changed:

- Move all utils / types / constants related to extensions to a new `@directus/extensions` package
- Allow reading all extensions on `/extensions` 

## Potential Risks / Drawbacks

- This PR is moving code around without changing the actual functionality
- People importing directly from `@directus/utils` / `@directus/types` / `@directus/constants` might have to import from `@directus/extensions` now. All types exported through `@directus/extensions-sdk` remain the same.

## Review Questions

- This is in prep for the bigger secure extensions implementation, and the foundation for #19921

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
@paescuj @azrikahar, gonna take an educated risk on this and merge it. I'd still like a brief retrospective review if possible, but less of a priority 👍🏻

**Review by azrikahar — COMMENTED** [review]
LGTM in general! Just some tiny questions/todos from what I can observe.

I believe we'll need to add a license file as well 👍

File: packages/extensions/package.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"funding": "https://github.com/directus/directus?sponsor=1",
+	"license": "MIT",
+	"author": "Rijk van Zanten <rijkvanzanten@me.com>",
+	"type": "module",
+	"exports": {
+		".": "./dist/shared/index.js",
+		"./node": "./dist/node/index.js",
+		"./package.json": "./package.json"
+	},
+	"main": "dist/index.js",
```
[the remark this answers is not in the document]
Comment (rijkvanzanten) [reply packages/extensions/package.json:20]: It should!

File: packages/extensions/readme.md
[the remark this answers is not in the document]
Comment (rijkvanzanten) [reply packages/extensions/readme.md:1]: Whoops! Copy paste issue

File: packages/extensions/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (paescuj) [reply packages/extensions/package.json:20]: Fix ready at https://github.com/directus/directus/pull/19926 😄

**Pull request merged**